### ISSUE TITLE
Inconsistent results while query and mutation in parallel from the same client

### ISSUE THREAD

**Issue by Interface007** [ChilliCream/graphql-platform#6729]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

I am using EFCore 8 with HotChocolate.Data 13.7.0 in an APS.NET Core project targeting .NET 8.0. 
On the client side, I use GraphQL.Client 6.0.0.

On the client side, I am requesting a "project" and locking that project in parallel:
``` C#
var projectTask = this.Client.GetQuery<Project>(projectId);
var locking = await this.Client.Lock<Project>(projectId);
var project1 = await projectTask;
var project2 = await this.Client.GetQuery<Project>(projectId);
```

`GetQuery` and `Lock` both create a query and call `IGraphQLClient.SendQueryAsync`. 

The "Project" entity is related to the "Task" entity and does have a property `public string ICollection<Task> Tasks { get; set; }`.
Both queries inlcude the `Tasks` property. `Lock` does update a single date field in the database by executing an explicit SQL statement.

On the server side, I have:
``` C#
[UsePaging(IncludeTotalCount = true)]
[UseProjection]
[UseFiltering]
[UseSorting]
public Task<IQueryable<Project>> GetProject([Service] IApplicationDbContext context, ClaimsPrincipal claimsPrincipal) => this.GetProject(context, claimsPrincipal);
```

`GetProject` creates an IQueryable including a where statement and `.Include(p => p.Tasks).AsSingleQuery()`.

In the client code (see above) the code that requests project1 starts, the code that requests locking starts then we wait for the locking task to be finished and after that for the project task. Wehn both have been finished, we start a new request with exactly the same query that has been used for the first project query.

Result: "sometimes" (in my project about 90% of the time), the response that fills `project1` does not include any `Tasks`. The identical request, that fills `project2` does always include the expected `Tasks`.

I logged the requests and responses by implementing a `System.Net.Http.DelegatingHandler` - both http-requests include `Tasks { ... }` in the query and the response includes '"Tasks": [],'.

I've added `.RegisterService<IApplicationDbContext>(ServiceKind.Resolver)` and event set `AddDbContext` to `ServiceLifetime.Transient` with no change in behavior.

I also added a `DbCommandInterceptor` to see whether there is any query accessing `Tasks` that returns 0 rows, but `result.HasRows` is always true (as expected). I have then added a Guid as a marker to the DbContext, so that I can see what queries each instance of a DbContext generates and for the GraphQL request that produces `Tasks.Count == 0`, there is no DB-Query that asks the db for the table Tasks. It seems like the 2nd, parallel request from the same client causes HotChocolate to not map the Tasks.

### Steps to reproduce

1. Query and mutate the an entity in parallel from the same client
 ``` C#
var projectTask = this.Client.GetQuery<Project>(projectId);
var locking = await this.Client.Lock<Project>(projectId);
var project1 = await projectTask;
var project2 = await this.Client.GetQuery<Project>(projectId);
```

### Version

13.7.0

**Comment by michaelstaib** [comment]
Hey There,

do you have a repro?

**Comment by Interface007** [comment]
Not yet - working on that and will add a comment as soon as I have something that I am allowed to share.

**Comment by michaelstaib** [comment]
Do we have a repro yet?

**Comment by Interface007** [comment]
The issue seems not to be easy to reproduce in an unrelated repo. I've done a simple example but was not able to reproduce the issue. So, it might be an edge case related to the specific query or the way we build our classes. We found a way to handle that issue on the client side, so I don't have a time budget from my customer to find the root cause of the issue.

________________________________
Von: Michael Staib ***@***.***>
Gesendet: Freitag, 15. Dezember 2023 08:02
An: ChilliCream/graphql-platform ***@***.***>
Cc: Sven Erik Matzen ***@***.***>; Author ***@***.***>
Betreff: Re: [ChilliCream/graphql-platform] Inconsistent results while query and mutation in parallel from the same client (Issue #6729)


Do we have a repro yet?

**Comment by michaelstaib** [comment]
I am closing this issue for now as not reproducible... If you find a rule to the issue please open a new issue and we will have a look at your problem.
